### PR TITLE
Fix ruleset filters not working correctly on song select

### PR DESCRIPTION
**Pull request by LumpBloom7**
Regression caused in this commit: https://github.com/ppy/osu/commit/02d8a6359a98a3c8ed393eda5c26376079d8cd79#diff-835d640b64dafd0c349eae448749a909626b2c02c31f9978b88cc3cfbc046605

Custom rulesets always have a negative onlineID, causing std beatmaps to always be filtered, and making custom rulesets unplayable for everybody. 

This PR changes the check to consider anything that isn't 0.

### CODE REVIEW

**Comment by peppy** [comment]
Could probably do with test coverage in `FilterMatchingTest`, since the commit you linked was also adding coverage there.

File: osu.Game.Tests/NonVisual/Filtering/FilterMatchingTest.cs [hunk 1]
```diff
@@ -78,6 +78,20 @@ public void TestCriteriaMatchingConvertedBeatmaps()
             Assert.IsFalse(carouselItem.Filtered.Value);
         }
 
+        [Test]
+        public void TestCriteriaMatchingConvertedBeatmapsForCustomRulesets()
+        {
+            var exampleBeatmapInfo = getExampleBeatmap();
+            var criteria = new FilterCriteria
+            {
+                Ruleset = new RulesetInfo { OnlineID = -25 },
```
Comment (peppy) [inline osu.Game.Tests/NonVisual/Filtering/FilterMatchingTest.cs:87]:
```suggestion
                Ruleset = new RulesetInfo { OnlineID = -1 },
```

No ruleset should be using anything but -1 (the automatic default) so the test should also specify this.

**Review by peppy — CHANGES_REQUESTED** [review]

**Comment by peppy** [comment]
I've fixed one more issue with the same code (was comparing `OnlineID` rather than `ShortName` for exact matching ruleset, which is very wrong).

Would appreciate a second review for sanity.

**Review by peppy — APPROVED** [review]

**Review by frenzibyte — APPROVED** [review]
Looks sane. While the `ShortName` changes make sense, I can see a few other places where rulesets are checked by `ID`/`OnlineID` instead (e.g. `BeatmapLeaderboard:L158`). We may want to look at updating those as well.

**Review by bdach — APPROVED** [review]
manually tested to be behaving as expected via ruleset template test projects

**Pull request merged**